network.go

```diff
@@ -808,7 +828,7 @@ func (n *network) updateSvcRecord(ep *endpoint, localEps []*endpoint, isAdd bool
 	}
 }
 
-func (n *network) getSvcRecords() []etchosts.Record {
+func (n *network) getSvcRecords(ep *endpoint) []etchosts.Record {
```

Comment:
Thinking of this function independently, could in future a caller be interested in all services on this network, regardless of any endpoint ?

If yes, we should allow caller to pass a nil `ep`, therefore we should skip the prefix check if nil was passed.